pkg/apis/flows/v1alpha1/parallel_conversion_test.go

```diff
@@ -0,0 +1,223 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/eventing/pkg/apis/flows/v1beta1"
+	messagingv1beta1 "knative.dev/eventing/pkg/apis/messaging/v1beta1"
+	"knative.dev/pkg/apis"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+)
+
+func TestParallelConversionBadType(t *testing.T) {
+	good, bad := &Parallel{}, &Parallel{}
+
+	if err := good.ConvertUp(context.Background(), bad); err == nil {
+		t.Errorf("ConvertUp() = %#v, wanted error", bad)
+	}
+
+	if err := good.ConvertDown(context.Background(), bad); err == nil {
+		t.Errorf("ConvertDown() = %#v, wanted error", good)
+	}
+}
+
+func TestParallelRoundTrip(t *testing.T) {
+	// Just one for now, just adding the for loop for ease of future changes.
+	versions := []apis.Convertible{&v1beta1.Parallel{}}
+
+	tests := []struct {
+		name string
+		in   *Parallel
+	}{{
+		name: "min configuration",
+		in: &Parallel{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:       "par-name",
+				Namespace:  "par-ns",
+				Generation: 17,
+			},
+			Spec: ParallelSpec{
+				Branches: []ParallelBranch{},
+			},
+		},
+	}, {
+		name: "full configuration",
```

Comment:
We started to also do tests where we start with v1beta1 -> v1alpha1 ->v1beta1, would you mind adding those tests as well? you can see for example pkg/apis/messaging/v1alpha1/channel_conversion_test.go.